drivers/modem/hl78xx/hl78xx_utility.c

```diff
@@ -0,0 +1,516 @@
+/*
```

Comment:
I am not really convinced about these utility functions. If they are useful, would it make sense to try to introduce them (or some of them) to other zephyr users (so in `include/zephyr/...` and in `lib/...`)?